### PR TITLE
[Limit orders] allowance check and approve

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

image

  # To Test
1. Approve for Limit order
2. Approve for Swap (old + new)
3. Approve for Eth flow

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @shoom3301 , I can't test as there is no PR preview link (the build failed).

**Comment by alfetopito** [comment]
> Hey @shoom3301 , I can't test as there is no PR preview link (the build failed).

Elena, keep in mind that you can also use the Vercel preview links like this one https://github.com/cowprotocol/cowswap/pull/1258#issuecomment-1293136585

File: src/cow-react/modules/swap/pure/SwapButtons/index.cosmos.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+function useCustomProps(): SwapButtonsContext {
+  const [swapButtonState] = useSelect('swapButtonState', {
+    options: Object.values(SwapButtonState),
+    defaultValue: SwapButtonState.NeedApprove,
+  })
+
+  return {
+    ...swapButtonsContext,
```
Comment (alfetopito) [inline src/cow-react/modules/swap/pure/SwapButtons/index.cosmos.tsx:32]: I don't know if this is something missing in the cosmos fixture props or on the button style itself.

The button styles are quite different:

Screen Shot 2022-10-27 at 16 59 42
Screen Shot 2022-10-27 at 16 59 34

See it [here](https://cowswap-prs-git-limit-orders-allowance-check-cowswap.vercel.app/cosmos/?fixtureId=%7B%22path%22%3A%22src%2Fcow-react%2Fmodules%2Fswap%2Fpure%2FSwapButtons%2Findex.cosmos.tsx%22%2C%22name%22%3A%22Default%22%7D)

File: src/cow-react/modules/shared/dumb/ApproveButton/index.cosmos.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { ApproveButton } from '.'
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply src/cow-react/modules/shared/dumb/ApproveButton/index.cosmos.tsx:1]: I will rename it in the next PR

File: src/cow-react/modules/swap/pure/SwapButtons/index.cosmos.tsx (hunk 1 above)
Comment (alfetopito) [reply src/cow-react/modules/swap/pure/SwapButtons/index.cosmos.tsx:32]: Ahhh I see.
I think we should unify all the styles.
Not for this PR though

**Pull request merged**